sccp/figures/figure4.py

```diff
@@ -6,40 +59,72 @@
     subplotLabel,
     getSetup,
     plotFactors,
-    plotProj,
 )
 from ..imports.scRNA import ThompsonXA_SCGenes
 from ..parafac2 import parafac2_nd
-from ..decomposition import plotR2X
-from ..crossVal import plotCrossVal
+from ..decomposition import R2XRank
+from ..crossVal import CrossValRank
+import pandas as pd
+import seaborn as sns
+
 
 
 def makeFigure():
     """Get a list of the axis objects and create a figure."""
     # Get list of axis objects
-    ax, f = getSetup((18, 25), (2, 4))
+    ax, f = getSetup((8, 10), (2, 2))
+    # ax, f = getSetup((9, 12), (2, 2))
 
     # Add subplot labels
     subplotLabel(ax)
 
     # Import of single cells: [Drug, Cell, Gene]
     data = ThompsonXA_SCGenes()
 
+    rank = 5
     _, factors, projs, _ = parafac2_nd(
         data,
-        rank=30,
-        random_state=1,
+        rank=rank,
+        random_state=0,
     )
 
-    flattened_projs = np.concatenate(projs, axis=0)
-    idxx = np.random.choice(flattened_projs.shape[0], size=200, replace=False)
+    # flattened_projs = np.concatenate(projs, axis=0)
+    # idxx = np.random.choice(flattened_projs.shape[0], size=200, replace=False)
 
-    plotFactors(factors, data, ax[0:2], reorder=(0, 2), trim=(2,))
+    # plotFactors(factors, data, ax[0:3], reorder=(0, 2), trim=(2,))
 
-    plotProj(flattened_projs[idxx, :], ax[3:5])
+    # plotProj(flattened_projs[idxx, :], ax[3:5])
 
-    # plotR2X(data, 30, ax[5])
+    r2x_error = R2XRank(data, rank+3)
 
-    # plotCrossVal(data.X_list, 30, ax[6], trainPerc=0.75)
+    cv_error = CrossValRank(data.X_list, rank+3, trainPerc=0.75)
+    
+    error = np.concatenate((r2x_error[0], r2x_error[1], cv_error[0], cv_error[1]))
```

Comment:
It's almost always better to keep these things separate. Concatenation introduces a whole bunch of very complicated indexing that frequently leads to errors.